### DISCUSSION TITLE
Features for Concept #11 - Finding an Image File and Displaying it [WAS concept 10]

### DISCUSSION THREAD

**Discussion by jcasman** [caitlynbakery/theta_image_save_9#3] [Ideas]
Use File_Picker - https://pub.dev/packages/file_picker

Pick an image from the same location gallery_saver saved it

Display image file within the Flutter app

**Comment by codetricity** [comment]
It's not clear if the file_picker can be used to display the file or if it is only picking the file.  We may not be able to display the file.

The README for file_picker indicates this:

> Load file data immediately into memory (Uint8List) if needed;

The result of picking the fle, may be one of two scenarios:

1. picker asks user to pick another app to display the file
2. picker can store the result of the pick action into a byte array

In order to accomplish the third feature, "display image file in Flutter app", we will need to have the picker return a byte array and then use   Flutter `Image.memory` to get the bytes into the displayed image.

**Reply by caitlynbakery** [reply]
Alright thanks for the information. Here's the link to Image.memory example that I'll use if needed. https://stackoverflow.com/questions/57863935/how-to-display-an-image-memory

I will also research other file pickers that might display the image in the application.

**Comment by codetricity** [comment]
It also seems like Flutter has `Image.file`

https://www.woolha.com/tutorials/flutter-display-image-from-file-examples#:~:text=In%20Flutter%2C%20displaying%20an%20image,you%20can%20also%20use%20FileImage%20.

If you limit the tutorial to Android, you can find the path of the file.

https://stackoverflow.com/questions/49835623/how-to-load-images-with-image-file

this has become concept 11.

**Reply by caitlynbakery** [reply]
I'll look into both examples and go through the Woolha example during work today. I'll temporarily stop File Picker and get Image.file working.

**Reply by caitlynbakery** [reply]
I started a new Github repo for additional testing of the package `permission_handler` [here](https://github.com/caitlynbakery/testing_permissions). The `Image.file` requires the permission from `permision_handler` in order to access the storage files. However, it seems that the `Permission.storage` may not work with Android :( 
Permission.storage Example: 

![Screen Shot 2022-07-20 at 10 05 14 AM](https://user-images.githubusercontent.com/22283010/180043928-f8aed4a0-c50b-4184-bb9e-58f3315a1701.png)

Permission.photos Example:

![Screen Shot 2022-07-20 at 10 05 28 AM](https://user-images.githubusercontent.com/22283010/180043982-2dfddcbe-cc5a-4853-bd26-ff39b2eb2398.png)

**Comment by codetricity** [comment]
For the next app, consider a more narrow focus to display the contents of a single directory.  You will need to find out the contents of a directory and then have a loop to iterate through all the files.

https://stackoverflow.com/questions/50521274/how-to-get-a-list-of-files-from-the-directory-and-pass-it-to-the-listview

**Reply by caitlynbakery** [reply]
Ok, I'll create a new project for listing out images in a loop. This will be Concept 12!

**Reply by caitlynbakery** [reply]
Here's the link for the new Github project: https://github.com/caitlynbakery/theta_list_images_12

**Reply by codetricity** [reply]
watch this video for information on an example workflow.
https://impel.io/capture-app/

**Reply by caitlynbakery** [reply]
Looks cool! Thanks for sharing ;)
